app/models/queen.rb

```diff
@@ -1,9 +1,15 @@
 class Queen < Piece
 
+  def valid_move?(x, y)
+    unless super(x, y)
+      return false
+    end
+    x_position == x || y_position == y || (x_position - x).abs == (y_position - y).abs
```

Comment:
i see the `Math` is strong in you.

took me a while to get this, but it looks good and most importantly you have tests :100: